activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1829,6 +1839,39 @@ def check_if_method_has_arguments!(method_name, args, message = nil)
         end
       end
 
+      def process_select_args(fields)
+        fields.flat_map do |field|
+          if field.is_a?(Hash)
+            transform_select_hash_values(field)
+          else
+            field
+          end
+        end
+      end
+
+      def transform_select_hash_values(fields)
+        fields.flat_map do |key, columns_aliases|
+          case columns_aliases
+          when Hash
+            columns_aliases.map do |column, column_alias|
+              arel_column("#{key}.#{column}") do |attr_name|
+                return connection.quote_table_name(attr_name)
+              end.as(column_alias.to_s)
+            end
+          when Array
+            columns_aliases.map do |column|
+              arel_column("#{key}.#{column}") do |attr_name|
+                connection.quote_table_name(attr_name)
+              end
+            end
+          when String, Symbol
+            arel_column(key.to_s) do |attr_name|
+              return connection.quote_table_name(attr_name)
```

Comment:
```suggestion
              connection.quote_table_name(attr_name)
```

Reply:
@casperisfine , without return, here

```ruby
Post.select(foo: :post_title).take
```

instead of
```
ActiveRecord::StatementInvalid: Mysql2::Error: Unknown column 'foo' in 'field list'
```
we will receive:
```
NoMethodError: undefined method `as' for "`foo`":String

            end.as(columns_aliases.to_s)
               ^^^
```

is it okay?

Reply:
No that's not ok. You must find a proper way to cast that missing column to proper Arel AST, like `where` does. 

I won't have time to look at this tonight, but if you are stuck just leave a comment and I'll have a look first thing tomorrow.

Reply:
@casperisfine, no, not stuck at all, should be fixed now.